server/src/main/java/org/elasticsearch/action/admin/indices/create/CreatePartitionsRequest.java

```diff
@@ -21,62 +21,89 @@
 
 package org.elasticsearch.action.admin.indices.create;
 
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
 import org.elasticsearch.Version;
 import org.elasticsearch.action.support.master.AcknowledgedRequest;
 import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.io.stream.StreamOutput;
 
-import java.io.IOException;
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.List;
+import io.crate.metadata.PartitionName;
+import io.crate.metadata.RelationName;
 
 public class CreatePartitionsRequest extends AcknowledgedRequest<CreatePartitionsRequest> {
 
-    private final Collection<String> indices;
+    private final Collection<PartitionName> partitions;
 
-    /**
-     * Constructs a new request to create indices with the specified names.
-     */
-    public CreatePartitionsRequest(Collection<String> indices) {
-        this.indices = indices;
+    public CreatePartitionsRequest(Collection<PartitionName> partitions) {
+        if (partitions.isEmpty()) {
+            throw new IllegalArgumentException("Must have at least one partition");
+        }
+        this.partitions = partitions;
     }
 
-    public Collection<String> indices() {
-        return indices;
+    public Collection<PartitionName> partitions() {
+        return partitions;
     }
 
     public CreatePartitionsRequest(StreamInput in) throws IOException {
         super(in);
-        if (in.getVersion().before(Version.V_5_3_0)) {
-            // The only usage of jobId was removed in https://github.com/crate/crate/commit/31e0f7f447eaa006e756c20bd32346b2680ebee6
-            // Nodes < 5.3.0 still send UUID which is written as 2 longs, we consume them but don't create an UUID out of them.
-            in.readLong();
-            in.readLong();
+        if (in.getVersion().onOrAfter(Version.V_5_10_0)) {
+            RelationName relation = new RelationName(in);
+            int numPartitions = in.readVInt();
+            this.partitions = new ArrayList<>(numPartitions);
+            for (int i = 0; i < numPartitions; i++) {
+                int numValues = in.readVInt();
+                List<String> partitionValues = new ArrayList<>(numValues);
+                for (int j = 0; j < numValues; j++) {
+                    partitionValues.add(in.readOptionalString());
+                }
+                this.partitions.add(new PartitionName(relation, partitionValues));
```

Comment:
Why not moving this into a `new PartitionName(StreamInput in)`? Same for the writing...

Reply:
The optimization only works if the relation name is the same for all partition names. So this can't be put into `PartitionName(StreamInput in)` - but it also means that the `CreatePartitionsRequest` signature as it is is unsafe as one could pass in partition names with different relation names. So I'll have to change that to either: 

- accept `RelationName relation, List<...> partitionValues`
- Ensure `List<PartitionName>` contains only a single relationName that's the same across all partitions
- Stream full PartitionName per partition

I'd kinda tend to the first option because it will make the transport implementation eaiser with the new RelationMetadata

Reply:
Ah right, missed this. Sounds good.